metrics.nim

```diff
@@ -157,13 +171,16 @@ when defined(metrics):
     return collector.metrics
 
   proc toTextLines*(collector: Collector, metricsTable: Metrics, showTimestamp = true): seq[string] =
-    result = @[
-      "# HELP $# $#" % [collector.name, collector.help.processHelp()],
-      "# TYPE $# $#" % [collector.name, collector.typ],
-    ]
-    for labelValues, metrics in metricsTable:
-      for metric in metrics:
-        result.add(metric.toText(showTimestamp))
+    try:
+      result = @[
+        "# HELP $# $#" % [collector.name, collector.help.processHelp()],
+        "# TYPE $# $#" % [collector.name, collector.typ],
+      ]
+      for labelValues, metrics in metricsTable:
+        for metric in metrics:
+          result.add(metric.toText(showTimestamp))
+    except ValueError as e:
+      result = @["metrics: " & e.msg] # whatever
```

Comment:
No, let it crash.

Reply:
this is called from the http server - metrics should really not crash when someone makes a web request and there's no way to know what happens if an exception rips through the async stack - it's likely not exception safe - this way, the error is displayed on the web page and can be handled with calm

Reply:
Someone does make a web request, once per second usually, but it won't look at your error messages and it surely won't handle anything with calm.

This top-level strategy of "library should not crash" is completely wrong in this case, and it has nothing to do with some async stack in some thread. It has to do with API callers being responsible for doing the wrong things at runtime.

Those wrong things resulting in hidden error strings is another wrong thing.

Reply:
nonetheless, the library launches a thread that it's responsible for - including handling any errors in it such that it doesn't affect other threads - I don't see the issue here, really? it's fairly standard practise that individual requests are not allowed to crash the whole server

Reply:
Why do you assume that this string generation is only used in the HTTP server? It's also used when printing the registry, individual collector objects and when logging with Chronicles: https://github.com/status-im/nim-metrics/blob/master/metrics/chronicles_support.nim

Reply:
So, `ValueError` is produced when the metrics library was not able to render the metric value as a string?

If so, this should be indeed handled much closer to the server request handling loop. In Chronicles, it will be handled by the `try/except` surrounding each log statement.

Reply:
well, that's the thing: often you could get 90% of the good information and a single error - practically, logging and metrics are additive features - any information you get out of them is good - if I get stats on cpu and vm memory, but not rss size, that's strictly better than getting nothing - so the rationale here, and what I think chronicles should do as well, is to treat to-string-conversion per field - that's more useful to users

Reply:
consider html - this is why it works so well, because even if some random tag is unsupported by some random browser, it delivers the maximum amount of valuable information to the user - logging and metrics? same thing - they're best-effort